### DISCUSSION TITLE
post/ssh-with-tpm/

### DISCUSSION THREAD

**Discussion by giscus** [lyc8503/lyc8503#9] [Announcements]
# post/ssh-with-tpm/

一直觉得直接将 SSH 私钥以一个普通文件的形式放在 ~/.ssh 下是一件很不安全的事情. 一旦有任何软件出现漏洞, 或者用户不慎运行了恶意软件 (不要过于自信的认为这不可能发生在自己的电脑上, 有的时候一个 typo 就能让你不知不觉中执行恶意代码), 毫无防护的 SSH 密钥很容易泄漏.   也考虑过购买硬件密钥设备, 但一直带着物理的密钥总是挺繁琐的事情, 而且密钥本身体积不大对于我这种不

https://blog.lyc8503.site/post/ssh-with-tpm/

**Comment by pedoc** [comment]
这个pin密码复杂度好像有要求,我输入6位纯数字密码会提示不符合复杂度要求.但是查看组策略,没有进行任何自定义设置

**Reply by lyc8503** [reply]
这个 PIN 的复杂性要求是 `tpmvscmgr.exe` 这个命令的参数指定的, 可以 `tpmvscmgr.exe /help` 看下.

**Comment by ghost** [comment]


**Reply by lyc8503** [reply]
我这里用的还是传统的 vSC, WHfB 我当时简单尝试了下也没成功, 配置似乎很复杂... 只能靠你自己再研究研究了, ~~期待你的成果~~.

**Comment by Do1e** [comment]
这里也提供另一种解决方法吧，我个人使用的是自建的bitwarden保管密码和[ssh密钥](https://bitwarden.com/help/ssh-agent/)

**Reply by lyc8503** [reply]
确实，这样方便一些。不过我是传统派的私钥不通过网络传输，一个设备一个私钥，TPM 可以加强这种保证。

~~革新派的读者可以参考上面的方案。~~

**Comment by qaq112233** [comment]
感谢分享，很有趣

**Comment by UMU618** [comment]
这个方法更简单：https://blog.umu618.com/2024/09/21/umutech-protect-your-private-keys-on-windows/

**Comment by p3ddd** [comment]
现在微软已经建议迁移到 FIDO2 了，更方便
- https://learn.microsoft.com/zh-cn/windows/security/identity-protection/virtual-smart-cards/virtual-smart-card-overview
- https://www.openssh.org/txt/release-8.2

**Reply by UMU618** [reply]
> 现在微软已经建议迁移到 FIDO2 了，更方便
> 
> * https://learn.microsoft.com/zh-cn/windows/security/identity-protection/virtual-smart-cards/virtual-smart-card-overview
> * https://www.openssh.org/txt/release-8.2

这种没带手机时就麻烦。

**Reply by p3ddd** [reply]
并非，现在支持使用Windows Hello进行认证，指纹或PIN码就行，不需要额外的设备。
